### PR TITLE
Update documentation for Milvus. Adding Mac OS support part.

### PR DESCRIPTION
**Pull request by soothing-rain**

### CODE REVIEW

**Comment by soothing-rain** [comment]
Please ignore the non-MD files. @jaime0815

File: internal/core/CMakeLists.txt [hunk 1]
```diff
@@ -16,15 +16,23 @@
 
 cmake_minimum_required( VERSION 3.18 )
 
+if ( APPLE )
```
Comment (jaime0815) [inline internal/core/CMakeLists.txt:19]: set variables within core_build.sh and internal/core/build.sh

File: internal/core/CMakeLists.txt [hunk 2]
```diff
@@ -16,15 +16,23 @@
...
+endif ()
+
 add_definitions(-DELPP_THREAD_SAFE)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 message( STATUS "Building using CMake version: ${CMAKE_VERSION}" )
 
-project(core LANGUAGES C CXX)
+project(core)
 include(CheckCXXCompilerFlag)
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```
Comment (jaime0815) [inline internal/core/CMakeLists.txt:25]: if use APPLE,  don't use MACOS at other places, keep style is uniform

File: internal/core/build.sh [hunk 3]
```diff
@@ -155,6 +155,8 @@ CMAKE_CMD="cmake \
 -DMILVUS_WITH_PROMETHEUS=${WITH_PROMETHEUS} \
 -DMILVUS_CUDA_ARCH=${CUDA_ARCH} \
 -DCUSTOM_THIRDPARTY_DOWNLOAD_PATH=${CUSTOM_THIRDPARTY_PATH} \
+-DMILVUS_ENABLE_PROFILING=${SUPPORT_PROFILING} \
```
Comment (jaime0815) [inline internal/core/build.sh:158]: where will new variables be assigned value?

Comment (jaime0815) [reply internal/core/build.sh:158]: if not , why add these options

File: internal/kv/rocksdb/cwrapper/CMakeLists.txt [hunk 4]
```diff
@@ -21,9 +21,15 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 include( ExternalProject )
-set( ROCKSDB_VERSION "6.15.2" )
-set( ROCKSDB_SOURCE_URL
-    "https://github.com/facebook/rocksdb/archive/v${ROCKSDB_VERSION}.tar.gz")
+if ( APPLE )
+    set( ROCKSDB_VERSION "6.27.3" )
```
Comment (jaime0815) [inline internal/kv/rocksdb/cwrapper/CMakeLists.txt:25]: Please pay attention to the x86 mac ci pass

File: scripts/run_go_codecov.sh [hunk 5]
```diff
@@ -23,14 +23,25 @@ set -e
...
+# TODO: "-race" is temporarily disabled for Mac Silicon. Add back when available.
 echo "Running unittest under ./internal"
-for d in $(go list ./internal... | grep -v vendor); do
-    go test -race -v -coverpkg=./... -coverprofile=profile.out -covermode=atomic "$d"
-    if [ -f profile.out ]; then
-        sed '1d' profile.out >> ${FILE_COVERAGE_INFO}
-        rm profile.out
-    fi
-done
+if [[ "$(uname -s)" == "Darwin" ]]; then
```
Comment (jaime0815) [inline scripts/run_go_codecov.sh:28]: run_cpp_codecov.sh is usable for mac

File: internal/core/build.sh (hunk 3 above)
Comment (soothing-rain) [reply internal/core/build.sh:158]: It is brought from: https://github.com/milvus-io/knowhere/blame/main/build.sh.

File: internal/core/CMakeLists.txt (hunk 1 above)
Comment (soothing-rain) [reply internal/core/CMakeLists.txt:19]: APPLE is a cmake variable: https://cmake.org/cmake/help/v3.0/variable/APPLE.html

File: internal/core/CMakeLists.txt (hunk 2 above)
Comment (soothing-rain) [reply internal/core/CMakeLists.txt:25]: This line is stale.

File: internal/kv/rocksdb/cwrapper/CMakeLists.txt (hunk 4 above)
Comment (soothing-rain) [reply internal/kv/rocksdb/cwrapper/CMakeLists.txt:25]: Will check.

File: scripts/run_go_codecov.sh (hunk 5 above)
Comment (soothing-rain) [reply scripts/run_go_codecov.sh:28]: Yes it is.

File: internal/core/CMakeLists.txt [hunk 6]
```diff
@@ -16,15 +16,23 @@
...
+if ( APPLE )
     message(STATUS "==============Darwin Environment==============")
     set(MACOS TRUE)
+    set(CMAKE_CXX_FLAGS "-L/usr/local/opt/libomp/lib -I/usr/local/opt/llvm/include -I/usr/local/include -I/usr/local/opt/openblas/include -I/usr/local/opt/libomp/include -DBOOST_STACKTRACE_GNU_SOURCE_NOT_REQUIRED=1")
```
[the remark this answers is not in the document]
Comment (soothing-rain) [reply internal/core/CMakeLists.txt:35]: Done.

File: internal/core/build.sh (hunk 3 above)
Comment (jaime0815) [reply internal/core/build.sh:158]: so, the variable's default is missing?

**Pull request closed without merging**